core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -121,18 +151,24 @@ namespace kagome::consensus {
       case BabeState::WAIT_BLOCK:
         // TODO(kamilsa): PRE-366 validate block. Now it is problematic as we
         // need t know VRF threshold for validation. To calculate that we need
-        // to have weights of the authorities and to get it we need to have the
-        // latest state of a blockchain
-
-        // synchronize missing blocks with their bodies
+        // to have weights of the authorities and to get it we need to have
+        // the latest state of a blockchain
+
+        // add new block header and synchronize missing blocks with their bodies
+        if (auto add_res = block_tree_->addBlockHeader(announce.header);
+            not add_res) {
+          log_->info("Could not apply block number {}, reason {}",
+                     announce.header.number,
+                     add_res.error().message());
+        }
```

Comment:
```suggestion
        // synchronize missing blocks with their bodies
```
This should be removed after merge with master

Reply:
Master already merged. This code will be removed by patch